File: web/src/types/platform.ts
export interface PlatformItem {
  id: number;
  name: string;
  url: string;
  iconUrl?: string;
  sortOrder: number;
  status: "active" | "disabled";
  createdAt?: string;
  updatedAt?: string;
}

export interface CreatePlatformPayload {
  name: string;
  url: string;
  iconUrl?: string;
  sortOrder: number;
  status: "active" | "disabled";
}

export interface UpdatePlatformPayload {
  name?: string;
  url?: string;
  iconUrl?: string;
  sortOrder?: number;
  status?: "active" | "disabled";
}

File: app/db/models/platform.py
"""Browser shortcuts pushed to every desktop user."""

from __future__ import annotations

from datetime import datetime
from typing import Optional

from sqlalchemy import Integer, String, Text, UniqueConstraint
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import TABLE_ARGS, Base, DateTime6, IdType, utc_now


class Platform(Base):
    """A centrally-managed browser shortcut available to desktop users."""

    __tablename__ = "platform"
    __table_args__ = (
        UniqueConstraint("name", name="uq_platform_name"),
        {**TABLE_ARGS},
    )

    id: Mapped[int] = mapped_column(IdType, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    url: Mapped[str] = mapped_column(String(2048), nullable=False)
    icon_url: Mapped[Optional[str]] = mapped_column(Text, nullable=True, default=None)
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0, index=True)
    status: Mapped[str] = mapped_column(String(16), nullable=False, default="active", index=True)
    created_at: Mapped[datetime] = mapped_column(DateTime6, nullable=False, default=utc_now)
    updated_at: Mapped[datetime] = mapped_column(DateTime6, nullable=False, default=utc_now, onupdate=utc_now)
